magenta/music/midi_io.py

```diff
@@ -327,5 +327,9 @@ def sequence_proto_to_midi_file(sequence, output_file,
   pretty_midi_object = sequence_proto_to_pretty_midi(
       sequence, drop_events_n_seconds_after_last_note)
   with tempfile.NamedTemporaryFile() as temp_file:
-    pretty_midi_object.write(temp_file.name)
+    pretty_midi_object.write(temp_file)
+    # Before copying the file, flush any contents
+    temp_file.flush()
+    # And back the file position to top
+    temp_file.seek(0)
```

Comment:
Why is the seek necessary?

Reply:
Certainly, copy works fine without `seek(0)`.
 But in the Python, to execute `flush` and `seek(0)` before reloading (copying) the file seems to be necessary (see also [ref1](https://stackoverflow.com/questions/22127960/read-from-file-after-write-before-closing) and [ref2](https://stackoverflow.com/questions/3253258/what-is-the-best-way-to-write-the-contents-of-a-stringio-to-a-file)), and it is reasonable.

Following code is a simple code that proves the need of flush & seek.

```py
import shutil
import tempfile


with tempfile.NamedTemporaryFile() as t_1:
    with tempfile.NamedTemporaryFile() as t_2:
        t_2.write("test_test".encode("utf-8"))
        t_2.flush()
        t_2.seek(0)  # without seek(0), print result is []
        shutil.copyfileobj(t_2, t_1, -1)
    t_1.seek(0)  # back the position after copy
    print(t_1.readlines())

```

[`tf.gfile.Copy`](https://www.tensorflow.org/api_docs/python/tf/gfile/Copy) works without `flush` & `seek(0)` (Specifically, it is needed to execute `flush` **or** `seek(0)`). I tried to confirm the implementation of `tf.gfile.Copy`, [but it is a little hard to chase](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/lib/io/file_io.py#L371).

So I took certainty and reasonable implementation. If it is too much, I'm ok to remove the `seek`.

Reply:
It's up to you whether you want to keep the seek(0) or remove it.

Reply:
Thank you for the review!
`tf.gfile.Copy` works without `flush` if `seek(0)` is executed. So the behavior of `tf.gfile.Copy` is a little mysterious.

Because of this, I want to keep this implementation if you are ok. I added the comment about it.